packages/ui/package.json

```diff
@@ -14,12 +14,24 @@
    },
    "peerDependencies": {
       "@chakra-ui/react": ">= 1.8.3 < 2",
-      "@core-ds/primitives": ">2.4"
+      "@core-ds/primitives": ">2.4",
+      "@emotion/react": ">=11.0.0 <12.0.0",
+      "framer-motion": ">=3.0.0 <4.0.0 || >=4.0.0 <5.0.0 || >=5.0.0 <6.0.0 || >=6.0.0 <7.0.0",
+      "@emotion/styled": ">=11.0.0 <12.0.0",
+      "react-dom": ">=16.8.6 <17.0.0 || >=17.0.0 <18.0.0",
+      "@babel/core": ">=7.0.0 <8.0.0"
```

Comment:
I don't understand why several of these are peer dependencies _and_ dev dependencies. Could you explain it or point me to somewhere I can learn about it?

Reply:
Yup: Basically, peer dependencies say, "At runtime, let me have access to the copy of this thing which my parent is using" ("parent" here being the thing which has a dependency on this package). This is useful for libraries like React, where we want to all share a copy, because all the React-using pieces of the system need to interact with the same React instance.

However, this creates a problem when developing the package. During development, there's no parent package (because we're working on the package all on its own). So there's nobody to provide that peer dependency. Thus, it's typical to _also_ include the peer dependency as a dev dependency, so that while developing we can use the dev dependency copy. When installed from NPM, dev dependencies won't be installed, so those modules should have no effect at runtime.

That's the basic logic. All this is complicated by the monorepo environment, where a package can be simultaneously used as a dependency _and_ be being developed. From what we've seen, Rush seems to ignore the peer dependencies of packages within the monorepo (that is, where both the parent and the package live in the monorepo). This is troublesome, and we're still figuring out what to do about it.